util/topsql/collector/main_test.go

```diff
@@ -63,12 +63,9 @@ func TestPProfCPUProfile(t *testing.T) {
 	require.True(t, len(data) > 0)
 	require.Equal(t, []byte("sql_digest value"), data[0].SQLDigest)
 
-	// Test after disabled, shouldn't receive any data.
+	// Test disable then re-enable.
 	topsqlstate.DisableTopSQL()
 	time.Sleep(interval * 2)
-	require.Equal(t, 0, len(mc.dataCh))
```

Comment:
Could you explain why is this removed?

Reply:
This check actually is meanless, it doesn't affect anything.

Reply:
When we call `topsqlstate.DisableTopSQL()`, I think there may be unconsumed data in `dataCh`, so we can't guarantee that "re-enable" is in effect.

How about:
```go
	// Test disable then re-enable.
	topsqlstate.DisableTopSQL()
	time.Sleep(interval * 2)
	dataChLen := len(mc.dataCh)
	topsqlstate.EnableTopSQL()
	deltaLen := 0
	for i := 0; i < 10; i++ {
		t1 := time.Now()
		data = <-mc.dataCh
		require.True(t, time.Since(t1) < interval*4)
		if len(data) > 0 {
			deltaLen++
			if deltaLen > dataChLen {
				// Here we can ensure that we receive new data after "re-enable".
				break
			}
		}
	}
```

Reply:
It's not a good idea not to check `Test after disabled, shouldn't receive any data.`

Reply:
@zhongzc, the current implementation in corner case can't make sure it, and actually we don't need to make sure this behavior, it doesn't matter.

Reply:
> ```go
> require.True(t, deltaLen > dataChLen)
> ```

nice catch, done.